src/Hook.php

```diff
@@ -0,0 +1,240 @@
+<?php
+
+namespace Utopia;
+
+class Hook {
+
+    /**
+     * Description
+     *
+     * @var string
+     */
+    protected string $desc = '';
+
+    /**
+     * Parameters
+     *
+     * List of route params names and validators
+     *
+     * @var array
+     */
+    protected array $params = [];
+
+    /**
+     * Group
+     *
+     * @var array
+     */
+    protected array $groups = [];
+
+    /**
+     * Action Callback
+     *
+     * @var callable
+     */
+    protected $action;
+
+    /**
+     * Injections
+     *
+     * List of route required injections for action callback
+     *
+     * @var array
+     */
+    protected array $injections = [];
+
+    /**
+     * Add Description
+     *
+     * @param string $desc
+     * @return static
+     */
+    public function desc(string $desc): static
```

Comment:
I think it makes a lot of sense to call this `setDesc` and be more consistent with the `getter` - `setter` pattern